app/setup_handlers.go

```diff
@@ -4,13 +4,33 @@ import (
 	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/x/upgrade/types"
+
+	dispensationtypes "github.com/Sifchain/sifnode/x/dispensation/types"
 )
 
-const upgradeName = "0.9.12"
+const upgradeName = "Reward-Mint"
 
 func SetupHandlers(app *SifchainApp) {
 	app.UpgradeKeeper.SetUpgradeHandler(upgradeName, func(ctx sdk.Context, plan types.Plan) {
 		app.Logger().Info("Running upgrade handler for " + upgradeName)
+
+		mintAmount, ok := sdk.NewIntFromString("200000000000000000000000000")
+		if !ok {
+			panic("failed to convert mint amount")
+		}
+		mintCoins := sdk.NewCoins(sdk.NewCoin(app.StakingKeeper.BondDenom(ctx), mintAmount))
```

Comment:
We need a check to make sure the mint only happens on a single release.  If someone did a hotfix without removing this code, I think we'd mint another batch.